### ISSUE TITLE
Bug: Overwrite given Digits in Sudoku.

### ISSUE THREAD

**Issue by Skeeve** [SteffenBauer/PocketPuzzles#26]
How to reproduce:

1. select pencilmarker
2. select any digit
3. press and hold any given digit

Result: Given digit is overwritten.

**Comment by SteffenBauer** [comment]
I'd need to ask for more details, as I cannot yet reproduce exactly what is going on.

Can you provide a more detailed step-by-step description, like "a) I have a cell filled with 1 2 4 as pencil marks. b) I select the cell by short-pressing it c) I long-click the number button '5' on the button bar ... I expect to happen X, but Y happens instead"

I also have the suspicion that this might even be a feature and not a bug; I introduced so-called 'one-click fill' mechanism some time ago to make entering numbers more smoothly on e-Ink screens, but not sure if this is what happens here.

**Comment by Skeeve** [comment]
I'll try
1. Create ANY Sudoku.
2. Decide for ANY given cell shown.
3. Click the Pencil Marker function.
4. preselect any of the digits so that a click on an empty cell will set this preselected as a pencil mark. Make sure it's not the digit you decided for in step 2.
5. a long press on any empty cell will place the preselected digit. I hope I described well enough up to this point.
6. if you now press long on the cell you decided for in step 2, it is overwritten by the preselected digit.

If you want I can give screenshots as an example. But I'm currently solving a sudoku rated "extrem" and would either like to solve that first or you have to live with a cluttered image ;)

BTW: I can also write in German if that's easier.

**Comment by SteffenBauer** [comment]
Lets stay in english, as other people might also read these threads.

I actually could reproduce the bug, as I understand it. With this method, I can overwrite a pre-given, fixed number.

Just to be sure we are talking about the same bug: I could reproduce the bug also in the Unequal puzzle, but not in Mathrax and Towers, where also pre-given fixed numbers can be on the board. Could you check if you can reproduce the issue also in Unequal, and if it also doesn't occur in Mathrax and Towers?

Additionally: Do you mean with "Pencil Marker function" the `-/--` button directly to the left of the Undo button? Strictly speaking, this is not "Pencil Marker", but a general "Swap short/long click function" button.

**Comment by Skeeve** [comment]
Yes! That's exactly the button I meant and I seem to remember that I already made the same mistake some time ago.

I will test as soon I solved the Sudoku. I fear to loose it when I swap game. IIRC there is only one save state which gets lost when you swap games.

BTW and on a side note: You also once told me I should checkt the generated sudokus. Thanks for that. I really like the difficult ones. When rating them with "Magic Sudoku Solver" on my iPhone they usually get an extreme rating.
